azurelinuxagent/ga/agent_update_handler.py

```diff
@@ -322,49 +318,46 @@ def run(self, goal_state):
                 if "Missing requested version" in GAUpdateReportState.report_error_msg:
                     GAUpdateReportState.report_error_msg = ""
 
-            if requested_version == CURRENT_VERSION:
+            # Check if an update is allowed and update next upgrade times even if no new agent is available for upgrade
```

Comment:
if some reason requested version is empty or missing in the goal state. we still want to update with old logic

Reply:
i don't think that is the case, since the check is on the boolean flag, and if the boolean is true, the version cannot be missing

Reply:
Boolean flag is local to agent. If we enable the GA versioning but the version property is empty in GS for whatever reason (may be issue in CRP), in that case, we except version but didn't get, so we log the warning and continue with self-update.

Reply:
I don't think that behavior is correct. If CRP is indicating that the machine is enrolled on RSM updates, we should never fallback to self updates (unless the CRP flag changes, of course).

If the flag is True and the version is missing, then, as you point out, it may be an issue on CRP. We should then report this error and skip the update.

Reply:
Initially this flag is not there when we implemented. That time we decided to do like this because things are in flux and if issue in crp and if fix takes days to reach prod, till then we don't do the update. That's not good so we want to continue with self-update and report status with this msg as RSM update has error.

After all that discussion today we think we shouldn't update sure I'll change it. But do you think is this issue in current release?

I'll consider changing as part of RSM stuff.

Reply:
Please remove the code and add it when it is needed, or comment it out or mark it somehow. Otherwise this may be missed in a future review. Marking it facilitates reviews.

Reply:
added comment to revisit this.